src/itwinai/torch/trainer.py

```diff
@@ -1429,130 +1400,106 @@ def set_epoch(self, epoch: int) -> None:
-                continue
-
-            param_type = values.get("type")
-
-            if param_type == "choice":
-                train_loop_config[name] = tune.choice(values["options"])
+    def _set_run_config(self) -> None:
+        run_config = self.config.get("run_config", {})
 
-            elif param_type == "uniform":
-                train_loop_config[name] = tune.uniform(
-                    float(values["min"]), float(values["max"])
-                )
-
-            elif param_type == "quniform":
-                train_loop_config[name] = tune.quniform(
-                    values["min"], values["max"], values["q"]
-                )
-
-            elif param_type == "loguniform":
-                train_loop_config[name] = tune.loguniform(values["min"], values["max"])
-
-            elif param_type == "qloguniform":
-                train_loop_config[name] = tune.qloguniform(
-                    values["min"], values["max"], values["q"]
-                )
+        if not run_config:
+            print("WARNING: No RunConfig provided. Assuming local or single-node execution.")
 
-            elif param_type == "randint":
-                train_loop_config[name] = tune.randint(values["min"], values["max"])
+        try:
+            storage_path = Path(run_config.pop("storage_path")).resolve()
 
-            elif param_type == "qrandint":
-                train_loop_config[name] = tune.qrandint(
-                    values["min"], values["max"], values["q"]
+            if not storage_path:
+                print(
+                    "INFO: Empty storage path provided. Using default path 'ray_checkpoints'"
                 )
+                storage_path = Path("ray_checkpoints").resolve()
 
-            elif param_type == "lograndint":
-                train_loop_config[name] = tune.lograndint(values["min"], values["max"])
-
-            elif param_type == "qlograndint":
-                train_loop_config[name] = tune.qlograndint(
-                    values["min"], values["max"], values["q"]
-                )
+            self.run_config = ray.train.RunConfig(**run_config, storage_path=storage_path)
+        except AttributeError:
+            print(
+                "Could not set Run Config. Please ensure that you have passed the "
+                "correct arguments for it. You can find more information for which "
+                "arguments to set at "
+                "https://docs.ray.io/en/latest/train/api/doc/ray.train.RunConfig.html"
+            )
 
-            elif param_type == "randn":
-                train_loop_config[name] = tune.randn(values["mean"], values["stddev"])
+    def _set_train_loop_config(self) -> None:
+        train_loop_config = self.config.get("train_loop_config", {})
 
-            elif param_type == "qrandn":
-                train_loop_config[name] = tune.qrandn(
-                    values["mean"], values["stddev"], values["q"]
+        if train_loop_config:
```

Comment:
If you handle the else part first, which is just a print statement followed by (I believe) returning an empty dict, you can remove a lot of indentation here.